### DISCUSSION TITLE
Ability to customize what columns are shown in association context.

### DISCUSSION THREAD

**Discussion by MrJoy** [avo-hq/avo#802] [Feature requests]
<img width="1573" alt="image" src="https://user-images.githubusercontent.com/113625/162585341-c9788f13-c1ae-42a5-adb3-0b527a9c97f8.png">

In this view, you can see that I have vendor displayed on the items list.  I would prefer that column only be visible when on the standalone items resource view.  There's frequently cases where I want to not display certain fields in an embedded view, or display certain fields only in an embedded view.

One approach to handle this might be to be able to define multiple resource classes that represent a given model, with only one showing up on the sidebar and the remainder being explicitly referenced in the field definitions of other resources.

e.g.

```ruby
field :some_collection, as: :has_many, resource: SomeCollection::SpecialCaseResource
```

Not sure what a sustainable/maintainable structure for organizing resource classes would be under such an arrangement.

Another approach would be to expand `show_on` / `hide_on`, but I feel like that would leave your team with considerably more complex and hard to maintain code, as well as leading to more tangled code on my end.  It would have the advantage of minimizing duplication of application-side code in the typical case.

**Comment by adrianthedev** [comment]
I see what you mean. I think that your `resource: SomeCollection::SpecialCaseResource` is the best way forward.

I'll convert this to a ticket for better tracking.

**Comment by segiddins** [comment]
Relatedly, it would sometimes be nice to show a `has_one` the way a `belongs_to` is displayed (I guess I could make an `inline_has_one` field type for that...), or similarly to be able to show `has_many` as if they were `tags` with each tag being a link

**Reply by adrianthedev** [reply]
I think the `has_one` can be retrofitted with a `show_inline` option. It's a fair bit of work but doable if you'd like to contribute a PR.

With the `has_many`, you could do that if you use a "resourceable" `tag`. For example, if you use a `Tag` model and you associate it through a `has_many` or polymorphic association, then create a `TagResource` to go with it.
I remember that was possible. I helped a client do it.
